### PR TITLE
feat: Support for fatalError, assert, precondition

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

The SDK was not able to retrieve the messages from Swifts fatalError, assert, or
precondition. This is fixed now by getting the message from the crashInfoMessage
of the binary image of libswiftCore.dylib.

The changes to SentryCrash for accessing the crashInfoMessage are merged from https://github.com/kstenerud/KSCrash/pull/375. Special thanks to @GLinnik21, who did the PR, and @kstenerud, who reviewed the PR.

Previously
Screen Shot 2021-12-29 at 11 15 55

https://sentry.io/organizations/sentry-sdks/issues/2862440420/events/a61f6937edd24abc9f4c5284ca50868d/?project=5428557

Now
Screen Shot 2021-12-29 at 11 14 56

https://sentry.io/organizations/sentry-sdks/issues/2893407482/events/72b601c512ff44f9a8d6ccbe3a403e22/?project=5428557

## :bulb: Motivation and Context

Fixes GH-662

## :green_heart: How did you test it?
Unit tests, simulator, and real devices.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

## :crystal_ball: Next steps
Update feature list of Apple in docs https://docs.sentry.io/platforms/apple/guides/ios/

### CODE REVIEW

**Comment by GLinnik21** [comment]
`crashInfoMessage` was originally researched by @bamx23

**Comment by philipphofmann** [comment]
> `crashInfoMessage` was originally researched by @bamx23

💯 @bamx23, thanks a lot.

**Comment by bamx23** [comment]
@philipphofmann great to see that it's adopted by such a project!)

**Review by Swatinem — APPROVED** [review]

File: Sources/Sentry/SentryCrashReportConverter.m [hunk 1]
```diff
@@ -438,6 +439,61 @@ - (void)enhanceValueFromNotableAddresses:(SentryException *)exception
     }
 }
 
+/**
+ * Get the message of fatalError, assert, and precondition to set is as the exception value if the
```
Comment (brustolin) [inline Sources/Sentry/SentryCrashReportConverter.m:443]: l: is this phrasing right? It sounds weird to me.
```suggestion
 * Get the message of fatalError, assert, and precondition to set it as the exception value if the
```

**Review by brustolin — APPROVED** [review]
This will be a great adding to the SDK.

LGTM!! 🎉

**Pull request merged**